src/Core/src/Hosting/ImageSources/ImageSourceServiceCollectionExtensions.cs

```diff
@@ -6,15 +6,31 @@ namespace Microsoft.Maui.Hosting
 {
 	public static class ImageSourceServiceCollectionExtensions
 	{
+		/// <summary>
+		/// Registers an image service with the underlying service container via AddSingleton.
+		/// </summary>
+		/// <typeparam name="TImageSource">The image type to register for</typeparam>
+		/// <typeparam name="TImageSourceService">The service type to register</typeparam>
+		/// <param name="services">The service collection</param>
+		/// <returns>The service collection</returns>
 		public static IImageSourceServiceCollection AddService<TImageSource, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TImageSourceService>(this IImageSourceServiceCollection services)
 			where TImageSource : IImageSource
 			where TImageSourceService : class, IImageSourceService<TImageSource>
 		{
+#pragma warning disable RS0030 // Do not use banned APIs, the current method is also banned
```

Comment:
Should we add a overload that doesn't use this ? And use it ourselfs ?

Reply:
If we use the wrong one, it's a "banned API" and we'd get a build error. This one doesn't seem to be highly used, but I'm open.